packages/contracts/contracts/ERC2981ContractWideRoyalties.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import '@openzeppelin/contracts/utils/introspection/ERC165.sol';
+
+import './ERC2981Base.sol';
+
+/// @dev This is a contract used to add ERC2981 support to ERC721 and 1155
+/// @dev This implementation has the same royalties for each and every tokens
+abstract contract ERC2981ContractWideRoyalties is ERC2981Base {
```

Comment:
why do we need this contract or the contract above? why not add the methods below to `Passport.sol` directly? the method above looks like it was already added to it anyway

Reply:
I actually referred to [link](https://github.com/dievardump/EIP2981-implementation) which seems to be the best available implementation of eip-2981 according to multiple sources.

Reply:
Right and if we have multiple contracts that are ERC2981s, then it would make sense to abstract out common functionality.

As it stands though, if `Passport.sol` is the only contract that will use these methods, then we could keep it all on Passport.sol. I'm not sure if these extra contracts mean extra gas cost we'll have to pay on deploy